test/test_c10d.py

```diff
@@ -3098,6 +3098,31 @@ def test_invalid_nccl_blocking_wait_env(self):
         self._run_invalid_nccl_blocking_wait_env('2147483647')
         self._run_invalid_nccl_blocking_wait_env('4294967295')
 
+    @requires_nccl()
+    @skip_if_not_multigpu
+    def test_nccl_timeout(self):
```

Comment:
Any reason for adding this to `CommTest` instead of `TimeoutTest`?

Reply:
`TimeoutTest` is not a `MultiProcessTestCase`.

Reply:
I know the other NCCL tests are here as well, but reiterating that they all should be moved to a separate NCCL class instead. The CommTest class was intended to only hold tests for functions in `csrc/distributed/c10d/comm.cpp`.